index.html

```diff
@@ -3514,6 +3514,58 @@ <h3>Reserved Extension Points</h3>
         </p>
 
       </section>
+
+      <section class="normative">
+        <h3>Ecosystem Compatibility</h3>
+
+        <p>
+There are a number of digital credential formats that do not natively use the
+data model provided in this document, but are aligned with a number of concepts
+in this specification. At the time of publication, examples of these digital
```

Comment:
I appreciate the need to distinguish mDL and other tokens as "digital credentials" from "verifiable credentials". 

I think addressing this issue more directly would be better.
 
Maybe something like "the term verifiable credentials is often synonymous with the term digital credentials, however our specification only uses the term verifiable credential to refer to the JSON-LD based data model we define here...."

The word "digital" is kinda glossing over the key distinction between credential formats, which is indeed media types... and W3C only recognizes vc+ld+json / vc+ld+jwt, vc+ld+cose as media types for W3C Verifiable Credentials (with or without any proof).
 
The rest of our text basically says this, and the context references make it clear that W3C only defines verifiable credentials in terms of JSON-LD... But other organizations and the industry as a whole does not necessarily agree.... 

For example, I disagree that the term "verifiable credential" only applies to JSON-LD.... 

But I do agree that "vc+ld+json" is a standard serialization of a digital credential that requires JSON-LD, and is specified by W3C. 

I remain in favor of and excited for other media types built on top of `vc+` that support other securing or data model considerations... and that DO NOT require ANY RDF or JSON-LD, or mappings.
 
VCDM v1 included support for things that were not the JSON-LD data model, the now infamous broken JWT and CL signature formats we have since corrected in v2.

This correction was accomplished in 2 ways:

1. we defined media types to remove ambiguity
2. we tried to agree to mapping rules for media types for media types that are not verifiable credentials, but are defined by this working group.

In my opinion, this PR rewrites a critical part of the day resolution, because it requires mappings, or externally defined media types, which was a major problem in v1 and v1.1... 

And which we never got consensus to do, see:

- https://github.com/w3c/vc-data-model/pull/1101
- https://github.com/w3c/vc-data-model/pull/1100

The day 3 resolution specifically stated, that mappings are ONLY required for media types defined by __this working group__... but this text asserts that mappings are required by any external specification, to conform to this working groups data model... and that a "mapping" is the key to creating a "verifiable credential".... 

Let me crystal clear, "a mapping is not needed to create a verifiable credential" or "any other digital credential"...

A mapping is required to produce `vc+ld+json` from any other well defined media type... 

It might seem like these are the same things, but they are not.

One of them is precise and easy to achieve, the other is marketing grand standing, that attempts to place W3C and JSON-LD at the center of the digital credentials universe, and claims the exclusive right to refer to "verifiable credentials" as being "JSON-LD" things.... But you can see from the media type that we requested registration for that this is not the case... because:

`application/vc+ld+json` defines that there are `application/vc` with different structured suffix.

Its fine to state that a mapping is required to produce `application/vc+ld+json` from other registered media types.... 

Its not ok to assert that only a mapping to JSON-LD is required to create a "application/vc+" or a "verifiable credential" in the "general sense of the word".

Not correcting this, will lead to a lot of very bad credential formats, that are needlessly shacl'ed (pun intended) to RDF and JSON-LD, which are not very well adopted technologies and which have been a continuous source of contention since the inception of the work in v1.

All that being said, I don't think the current text regarding "digital credentials" and "verifiable credentials" is far off from something I would approve.

And my comment here is none blocking, but there are other blocking comments on the sections that follow.

Reply:
Thinking a bit more about this... I think `application/vc` is really a "claimset" media type... that does not assume any serialization or securing format...

`application/vc+ld+json`  assumes JSON-LD serialization, and no securing format...

`application/vc+ld+jwt`  assumes JSON-LD serialization, and JWT securing format...

The use of structured suffixes is intentional extensibility, and we should say that directly.

Reply:
I think the trouble of "vc+ld+json" vs "vc+ld+jwt" is that they are completely different things. Other one is data model, another one is digital signature for any content.

I don't think a thing like `application/vc+ld+jwt` should even exist, as if you look into JWS spec, the `typ` is media type for the complete JWS and `cty` is for the JWS Payload. How it is transported over HTTP is a third media type.

Anyway, listing ecosystem support for signature schemes or a like should be transformed into "their corresponding data models". This can also be implicitly achieved by stating that the transformation algorithm input is data model and output is data model.

Reply:
> I don't think a thing like application/vc+ld+jwt should even exist, as if you look into JWS spec, the typ is media type for the complete JWS and cty is for the JWS Payload. How it is transported over HTTP is a third media type.

Yes, agreed, not to mention that IF we were to have something like this, it would have to be `application/vc+ld+json+jwt` to make any sense from a structured-suffix processing standpoint. I've also seen `application/vc+jwt` floating around, which also doesn't make sense as `application/vc` is a meta-model (at best) with no serialization.

Still mulling the rest of @OR13 input, but wanted to highlight that @mtaimela's feedback resonated with me, glad to know I'm not the only one thinking that.

Reply:
Great discussion.

There are several layers (inside out)
- data model (claim set)
- vocabulary (as defined in the specification and the @context)
- format (defines also processing elements)
- securing/protection (also defines a format/serialisation)

Data model and vocabulary are the core of the specification and if I understand correctly, most of the community agrees with it.
Format: JSON-LD, JSON (please continue reading)
Protection: JWS, Data Integrity Proofs, ...
Other processing elements (JSON-LD features, internationalisation, representation, ...)

As mentioned by @mtaimela we need to distinguish between different layers.
If VC is protected, the media type is defined by the protecting/securing mechanism.
The signature needs to define how to learn about the content/payload type.

In the case when a payload is protected, we'll have:
1. media type as defined by the signature (JOSE, ...) - this will usually be the media type in the HTTP header (when transported via HTTP)
2. signature type (good example is JAdES that defines several JWS profiles) - This tells the user how to process the signature and where/how the type is defined depends on the media type from point 1
3. content or payload type (VC+JSON+LD, ...) - we need to tell the user what is the payload and how to process it

Payload may always be processed as JSON or JSON-LD, depending on the use case and requirements. 

It is important to know that we have 3 signature formats
- enveloping (e.g., JSON serialised JWS)
- enveloped (e.g., Data Integrity Proofs)
- detached (e.g., compact serialised JWS)

enveloping and detached will carry the payload inside of the signature, whereas the enveloped signature carries the signature inside of the payload.  

Processing the payload: whether the payload needs to be processed as JSON or JSON-LD depends on what/how we want to match or process the claims. This information is important when requesting or presenting VCs as all actors need to agree on the processing rules of the payload.

Reply:
> `application/vc+ld+json` defines that there are `application/vc` with different structured suffix.

This is a complete misunderstanding and misstatement of how media types work.

Reply:
@OR13 
**Topic 1:**
`application/vc+ld+json+jwt` is very forcefully built and not fully aligning with the JWS specs. JWS defines 2 properties for media types to define itself with `typ`, and JWS payload with `cty`. The transport media is same as `typ`, but that one is lost after transport. There is no need to hack the `typ` with extensions.

if you want JWS to be with JWT profile, then `typ` is the "JWT", which is just uppercase and shortened version of application/jwt.

If you want to inform that the JWS Payload is `vc+ld+json`, then you should use the `cty` property for it. JWS and JWE specs specifies this more accurately, while JWT spec is just saying "everything is json so it's all the same media type, not recommended to use as it's json anyway".

TL;DR;
`application/vc+ld+json+jwt` is just `application/jwt` which defines its payload to be `application/vc+ld+json` with `cty` property, so there is no `application/vc+ld+json+jwt`.

Please keep the modular boundaries in mind.
- Signature schemes sign payloads (JWS / linked proofs / JAdES)
- Data models are the payloads (VCDM).
- Media types informs the how to parse stuff and is for both

**Topic 2:**
> application/vc can be used without +ld+json for example as application/vc+ld+cbor.

If `application/vc` is a meta model, then the documented **transformation algorithm** can cast it into any they like, even without LD. `application/vc+sd-jwt` on the other hand is breaching the boundaries of data model and signature scheme and should not exist at all.

Reply:
Regarding why use `typ`... (also discussed at the face to face, and on various issues)... https://datatracker.ietf.org/doc/html/draft-ietf-oauth-jwt-bcp-07#section-3.11

> if you want JWS to be with JWT profile, then typ is the "JWT", which is just uppercase and shortened version of application/jwt.

This has been discussed, and is considered a mistake, see the JWT BCP linked above, it was one of the first things the working group fixed related to VC-JWT... it leads to token confusion attacks.

> If you want to inform that the JWS Payload is `vc+ld+json`, then you should use the `cty` property for it.

Not true, see the JWT BCP... of course if you are not making a "token" then yea just securing a vanilla content type using `cty` works... Not sure if OIDC supports such formats (or JSON serialized JadEs stuff), so that trade off could have other consequences for adoption.

> TL;DR;
> `application/vc+ld+json+jwt` is just `application/jwt` which defines its payload to be `application/vc+ld+json` with `cty` property, so there is no `application/vc+ld+json+jwt`.

^ this implies there is no value in structured suffixes, or the JWT BCP... I see both as valuable... However only the JWT BCP is an RFC, multiple suffixes is still a draft, see: https://datatracker.ietf.org/doc/draft-ietf-mediaman-suffixes/

> JWT does not imply JSON, JWT is base64url encoded strings separated with dots.

This is not true, see also this thread on the "CWT" side of this same issue:

- https://mailarchive.ietf.org/arch/msg/cose/KNRge3vxXF3A2LY24DNo6ZQxrNc/

>  Only the JWTs inner payload is implied to be JSON, media type parsers are lost with the +json notation when encountering jwt.

JWT headers are also JSON, it is true that they are often discarded after verification.

I am not sure I follow what you are saying, but `+jwt` is used as as structured suffix for several media types already: 

(search +jwt) ... https://www.iana.org/assignments/media-types/media-types.xhtml

Are you saying that you don't know if their payloads are json because they didn't use multiple suffixes (currently not an RFC, which is part of why they don't do this)... For example, every +jwt registered media type could have been `+json+jwt`... if multiple suffixes were an RFC... ?

Since every +jwt secures a JSON `cty`... hence the choice to omit `cty` when `typ` is provided in most (all?) the existing registered media types that use `+jwt`.

(everything I am saying also applies to `+sd-jwt` btw).


> Please keep the modular boundaries in mind.
> 
> * Signature schemes sign payloads (JWS / linked proofs / JAdES)
> * Data models are the payloads (VCDM).
> * Media types informs the how to parse stuff and is for both

Correct, media types can be used via `typ` and/or `cty`.... or `content_type` in COSE, or `Content-Type` and `Accept` in HTTP headers.

It's valuable to be precise to avoid "confusion attacks".

> If `application/vc` is a meta model, then the documented **transformation algorithm** can cast it into any they like, even without LD. 

There is no requirement to record a "transformation algorithm" to register a media types... W3C does not control the media types registry (or HTML, or URL standards for that matter).

I don't know if anyone will ever register `application/vc` by itself... that will be up to experts to review and approve... I suspect they would reject it, if it had no processing requirements that worked without assuming a serialization.... for example.. `application/jwt` assumes JSON and base64url encoding... but `application/vc` does not... it offers at least `vc+ld+json` which is `+ld+json` which is `+json` ... meaning you can treat the entire thing as JSON... similar to how you know that `+jwt` secures JSON claimsets... which is literally what verifiable credentials are:

> A [credential](https://w3c.github.io/vc-data-model/#dfn-credential) is a set of one or more [claims](https://w3c.github.io/vc-data-model/#dfn-claims) made by the same [entity](https://w3c.github.io/vc-data-model/#dfn-entities).

- https://w3c.github.io/vc-data-model/#credentials

> `application/vc+sd-jwt` on the other hand is breaching the boundaries of data model and signature scheme and should not exist at all.

"the boundaries of data model "... JSON-LD is JSON, see https://www.w3.org/TR/json-ld11/#relationship-to-rdf

> JSON-LD is a [concrete RDF syntax](https://www.w3.org/TR/rdf11-concepts/#dfn-concrete-rdf-syntax) as described in [[RDF11-CONCEPTS](https://www.w3.org/TR/json-ld11/#bib-rdf11-concepts)]. Hence, a JSON-LD document is both an RDF document and a JSON document and correspondingly represents an instance of an [RDF data model](https://www.w3.org/TR/rdf11-concepts/#data-model).

According to JSON-LD itself... JSON-LD is JSON, which makes securing it with `+jwt` or `+sd-jwt` both valid... since both structured suffixes secure JSON... further, the use of structured suffixes DIRECTS processors to treat the media type as JSON.... so beyond using JWTs that work on JSON, you can also use JSON Schema which works on JSON, or JSON Pointer, which also works on JSON.

Since you mention JAdES... Its true, that `+jwt` is a structured suffix that does not apply to JSON serializations of a JWS.

If it's important to be able to smuggle data in the unprotected headers of a JWS in JSON serialization, it is true that `+jwt` does not support that use case afaik... but afaik, `+sd-jwt` does support that kind of thing... as does `+cose`.... 

- https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-05#name-jws-json-serialization
- https://www.rfc-editor.org/rfc/rfc9052#section-2

`+cwt` would not be appropriate since `vc+ld+json` is not CBOR, but assuming `vc+ld+cbor` existed, and was used as a CWT claimset, `+cwt` would be a structured suffix that could secure it.

Reply:
Hi Orie, thank you for the reply, very insightful and much appreciated. I stand corrected on the JWT explicit typing, didn't know that JWT deviates that much when compared to JWS/JWE. I still don't see the value of tokenizing VCs, as you will always introduce a VP when disclosing those, so they are never standalone (maybe some coupon case could exist without holder binding), but this topic is very much outside of this PR 😇.

To not derail the original discussion too much, could you please comment on how you see the transformation algorithm input/output?. I see that unlocking this question, will unlock the rest.

Bi- or uni-directional transformations:
1. Any Data Model -> VCDM (this spec)
2. Any Signature scheme -> VCDM (this spec)

This is quite important, as with option 1 all signature options must have a **source data model** to transform from, into `vc+ld+json` (which is then processed). The **source data model** could then be used for signature purposes how they wish. This follows the normal boundaries each signature scheme has and allows explicit media types for all tents.

If option 2 is allowed, then we see transformers that converts data model into signed data model, and this most of the time violates other signature schemes (like JAdES).

This question will also impact the following line
> At the time of publication, examples of these digital credential formats include JSON Web Tokens (JWTs), CBOR Web Tokens (CWTs), ISO-18013-5:2021 (mDLs), AnonCreds, Gordian Envelopes, and Authentic Chained Data Containers (ACDCs). 

As then the securing of credentials, like JWT, could be removed from VCDM. These are not data models, but "signature schemes" to secure the data models. This also solves the problem of identifying the media types which should be used, and will push them to the owners of the tents.

Reply:
Replying on this megathread, because it's where this problematic comment was made:

> [@OR13] Thinking a bit more about this... I think `application/vc` is really a "claimset" media type... that does not assume any serialization or securing format...

`application/vc` is NOT a media type, at all, noway, nohow, unless a registration was submitted to IANA out-of-band and out-of-WG and out-of-CG, in which case we cannot proceed with anything to do anything with it until we have a copy of the relevant spec.

Reply:
I believe this megathread can/should be solved under the vc-jose-cose, however we're maybe very close to a solution.
 
Main question is actually about JWT (not JWS, usage and role of JWS are clear). 

This is my view and I might be wrong (but this is how I understand the JWT RFC - JWT is a profile for JWS: compact serialised + JWT claims that need to be defined by the use case):

If JWT is seen as a securing mechanism that uses/adds additional JWT claims to the payload to protect the content, then all the issues and conflicts can easily be resolved. In a conceptual model we should be able to distinguish between an issuer and a signer. In most cases issuer == signer.

If my understanding of the JWT is wrong (if so, let me know) it would be good to clarify the position and usage of JWT.

@selfissued @OR13 can you help with this one or if there's anyone else in the group who can help. Thank you!

Reply:
Please open an issue on vc-cose-jose and move this discussion there. If there is anything left in this thread that applies to this PR, please make concrete change suggestions.

Reply:
@msporny Moved it to https://github.com/w3c/vc-jose-cose/issues/132. This thread should not block this PR.